### ISSUE TITLE
Essential gem dependencies are not defined correctly

### ISSUE THREAD

**Issue by kwallaschek** [moneyforward/ruby_branch_coverage#7]
After introducing ruby_branch_coverage to ca_web, we encountered problems building the application.

The problem was that the dependencies were not defined correctly.
For example: https://github.com/moneyforward/ruby_branch_coverage/blob/b6eb9b12fc6a2f93c42f1a58b76cb8d42aacf85e/Gemfile#L7

Now, in ca_web, we only have `simplecov` in the testing group and not in every environment.

`ruby_branch_coverage` was introduced to all environments (not just test), which made it impossible to build the application since this gem relied on a gem that it did not specify.

# Wanted fix
Please correctly define all dependencies

<Summary>
Full Error Log
<Details>

```
#26 [builder 7/7] RUN --mount=type=secret,id=secret_key_base   export CA_SECRET_KEY_BASE=$(cat /run/secrets/secret_key_base) &&   RAILS_ENV=production bundle exec rake assets:precompile assets:clean &&   npm run clean &&   npm run build &&   bundle exec bootsnap precompile
#26 7.678 rake aborted!
#26 7.686 LoadError: cannot load such file -- simplecov
#26 7.686 /mbook/app/rails/ca_web/vendor/bundle/ruby/3.0.0/gems/bootsnap-1.10.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:27:in `require'
#26 7.686 /mbook/app/rails/ca_web/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
#26 7.686 /mbook/app/rails/ca_web/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.3.2/lib/active_support/dependencies.rb:332:in `block in require'
#26 7.686 /mbook/app/rails/ca_web/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.3.2/lib/active_support/dependencies.rb:299:in `load_dependency'
#26 7.686 /mbook/app/rails/ca_web/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.3.2/lib/active_support/dependencies.rb:332:in `require'
#26 7.686 /mbook/app/rails/ca_web/vendor/bundle/ruby/3.0.0/gems/ruby_branch_coverage-0.1.4/lib/tasks/simplecov.rake:3:in `<main>'
#26 7.686 /mbook/app/rails/ca_web/vendor/bundle/ruby/3.0.0/gems/bootsnap-1.10.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:48:in `load'
#26 7.686 /mbook/app/rails/ca_web/vendor/bundle/ruby/3.0.0/gems/bootsnap-1.10.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:48:in `load'
#26 7.686 /mbook/app/rails/ca_web/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.3.2/lib/active_support/dependencies.rb:326:in `block in load'
#26 7.686 /mbook/app/rails/ca_web/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.3.2/lib/active_support/dependencies.rb:299:in `load_dependency'
#26 7.686 /mbook/app/rails/ca_web/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.3.2/lib/active_support/dependencies.rb:326:in `load'
#26 7.686 /mbook/app/rails/ca_web/vendor/bundle/ruby/3.0.0/gems/railties-6.1.3.2/lib/rails/engine.rb:675:in `block in run_tasks_blocks'
#26 7.686 /mbook/app/rails/ca_web/vendor/bundle/ruby/3.0.0/gems/railties-6.1.3.2/lib/rails/engine.rb:675:in `each'
#26 7.686 /mbook/app/rails/ca_web/vendor/bundle/ruby/3.0.0/gems/railties-6.1.3.2/lib/rails/engine.rb:675:in `run_tasks_blocks'
#26 7.686 /mbook/app/rails/ca_web/vendor/bundle/ruby/3.0.0/gems/railties-6.1.3.2/lib/rails/application.rb:520:in `block in run_tasks_blocks'
#26 7.686 /mbook/app/rails/ca_web/vendor/bundle/ruby/3.0.0/gems/railties-6.1.3.2/lib/rails/engine/railties.rb:15:in `each'
#26 7.686 /mbook/app/rails/ca_web/vendor/bundle/ruby/3.0.0/gems/railties-6.1.3.2/lib/rails/engine/railties.rb:15:in `each'
#26 7.686 /mbook/app/rails/ca_web/vendor/bundle/ruby/3.0.0/gems/railties-6.1.3.2/lib/rails/application.rb:520:in `run_tasks_blocks'
#26 7.686 /mbook/app/rails/ca_web/vendor/bundle/ruby/3.0.0/gems/railties-6.1.3.2/lib/rails/engine.rb:464:in `load_tasks'
#26 7.686 /mbook/app/rails/ca_web/vendor/bundle/ruby/3.0.0/gems/railties-6.1.3.2/lib/rails/railtie.rb:207:in `public_send'
#26 7.686 /mbook/app/rails/ca_web/vendor/bundle/ruby/3.0.0/gems/railties-6.1.3.2/lib/rails/railtie.rb:207:in `method_missing'
#26 7.686 /mbook/app/rails/ca_web/Rakefile:7:in `<top (required)>'
#26 7.686 /mbook/app/rails/ca_web/vendor/bundle/ruby/3.0.0/gems/rake-13.0.6/exe/rake:27:in `<top (required)>'
#26 7.686 /usr/local/bundle/gems/bundler-2.2.25/lib/bundler/cli/exec.rb:63:in `load'
#26 7.686 /usr/local/bundle/gems/bundler-2.2.25/lib/bundler/cli/exec.rb:63:in `kernel_load'
#26 7.686 /usr/local/bundle/gems/bundler-2.2.25/lib/bundler/cli/exec.rb:28:in `run'
#26 7.686 /usr/local/bundle/gems/bundler-2.2.25/lib/bundler/cli.rb:481:in `exec'
#26 7.686 /usr/local/bundle/gems/bundler-2.2.25/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
#26 7.686 /usr/local/bundle/gems/bundler-2.2.25/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
#26 7.686 /usr/local/bundle/gems/bundler-2.2.25/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
#26 7.686 /usr/local/bundle/gems/bundler-2.2.25/lib/bundler/cli.rb:31:in `dispatch'
#26 7.686 /usr/local/bundle/gems/bundler-2.2.25/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
#26 7.686 /usr/local/bundle/gems/bundler-2.2.25/lib/bundler/cli.rb:25:in `start'
#26 7.686 /usr/local/bundle/gems/bundler-2.2.25/exe/bundle:49:in `block in <top (required)>'
#26 7.686 /usr/local/bundle/gems/bundler-2.2.25/lib/bundler/friendly_errors.rb:128:in `with_friendly_errors'
#26 7.686 /usr/local/bundle/gems/bundler-2.2.25/exe/bundle:37:in `<top (required)>'
#26 7.686 /usr/local/bundle/bin/bundle:23:in `load'
#26 7.686 /usr/local/bundle/bin/bundle:23:in `<main>'
#26 7.686 (See full trace by running task with --trace)
#26 ERROR: process "/bin/sh -c export CA_SECRET_KEY_BASE=$(cat /run/secrets/secret_key_base) &&   RAILS_ENV=production bundle exec rake assets:precompile assets:clean &&   npm run clean &&   npm run build &&   bundle exec bootsnap precompile" did not complete successfully: exit code: 1
```

</Details>
</Summary>

**Comment by visu-suganya** [comment]
Hi @kwallaschek 
The reason for the issue is: simplecov was not defined as run time dependency.
Now it is fixed and [tested](https://github.com/moneyforward/tax_return_api/pull/4226) and docker build success after adding simplecov as run time dependency for this library

Released in [V0.1.5](https://github.com/moneyforward/ruby_branch_coverage/pkgs/rubygems/ruby_branch_coverage). Thank you.

**Comment by visu-suganya** [comment]
@kwallaschek 
Please check on ca_web and let me know. I will close this issue.
Thank you
